app/src/main/java/org/eyeseetea/malariacare/data/database/model/Survey.java

```diff
@@ -1006,10 +1008,12 @@ public void updateSurveyStatus() {
             return;
         }
 
-        SurveyAnsweredRatio answeredRatio = this.reloadSurveyAnsweredRatio();
+        AReloadSurveyAnsweredRatioUseCase reloadSurveyUseCase =
```

Comment:
@manuelplazaspalacio an UseCase can not be invoked from data layer, Use case should be invoked only from presentation layer. On this way updateSurveyStatus has more of one resposability. A possible option is from where we invoke updateSurveyStatus first we calculate SurveyAnsweredRatio and pass it as parameter to updateSurveyStatus